Mlem/API/Requests/SearchRequest.swift

```diff
@@ -37,9 +37,9 @@ struct SearchRequest: APIGetRequest {
         creatorId: Int? = nil,
         limit: Int? = nil
     ) {
-        self.instanceURL = account.instanceLink
-        var queryItems: [URLQueryItem] = [
-            .init(name: "auth", value: account.accessToken),
+        self.instanceURL = account?.instanceLink ?? DefaultLemmyServer
+        let queryItems: [URLQueryItem] = [
+            .init(name: "auth", value: account?.accessToken),
```

Comment:
Don't make null-tolerant versions of these APIs, make additional .init's which take only the Instance URL, so that the client a layer up can provide it that default URL if it wishes so

Else this will most likely just lead to buggy and inconsistent behaviour

Reply:
I wanted to leave the UI layer blissfuly unaware when the account is nil and having the API layer have a sensible fallback seemed like the way to go, 
I'm not really sure how to implement this cleanly in the UI layer without adding this parameter, and passing it down (with a fallback in case the account is nil) on every request, which to me looks dirty, but if you think it's the way I would be happy to change it 😄